### ISSUE TITLE
Implementation of "memset" function from the standard C library

### ISSUE THREAD

**Issue by shmarovfedor** [esbmc/esbmc#367]
ESBMC does not find a bug in the following code:

```
typedef unsigned int __kernel_size_t;
typedef __kernel_size_t size_t;
typedef unsigned int __u32;

struct compstat {
   __u32 unc_bytes ;
   __u32 unc_packets ;
   __u32 comp_bytes ;
   __u32 comp_packets ;
   __u32 inc_bytes ;
   __u32 inc_packets ;
   __u32 in_count ;
   __u32 bytes_out ;
   double ratio ;
};

struct ppp_comp_stats {
   struct compstat c ;
   struct compstat d ;
};

int main() {
  struct ppp_comp_stats cstats ;
  void *point;
  point = (void *)(& cstats);
  memset(point, 0, 81UL);
  return 0;
}
```
Here `memset` is trying to write past the last member of the struct `cstats` which has a size of 80 bytes.

**Comment by mikhailramalho** [comment]
This is again our code that calculates the size of a struct but rounds it up to word size before checking for bounds violation.

I plan to have another go at it during the weekend and will update here when I get something working.

**Comment by shmarovfedor** [comment]
I see what you mean. I just checked, and ESBMC starts giving the correct answer only when `N>145` in `memset(point, 0, N);`. So it makes sense I guess cause the rounding up makes the struct `cstats` 144 bytes. Thanks for the explanation, @mikhailramalho !!

**Comment by lucasccordeiro** [comment]
@shmarovfedor: could you please check if this issue was solved?

**Comment by fbrausse** [comment]
This issue has been solved with the same fix as #331.
